mdal/frmts/mdal_mike21.hpp

```diff
@@ -0,0 +1,104 @@
+/*
+ MDAL - Mesh Data Abstraction Library (MIT License)
+ Copyright (C) 2023 Lutra Consulting Ltd.
+*/
+
+#ifndef MDAL_MIKE21_HPP
+#define MDAL_MIKE21_HPP
+
+#include <string>
+#include <memory>
+#include <regex>
+
+#include "mdal_data_model.hpp"
+#include "mdal_memory_data_model.hpp"
+#include "mdal.h"
+#include "mdal_driver.hpp"
+
+namespace MDAL
+{
+  class MeshMike21: public MemoryMesh
+  {
+    public:
+      MeshMike21( size_t faceVerticesMaximumCount,
```

Comment:
is it necessary to use this new derived class ? I don't see in the code where the methods of this class are used.

Reply:
This just copies the approach used in 2DM driver and it should allow the use of the mesh structure later. No direct use for those fuctions right now. But for 2DM driver it is rather similar, those functions are only used in `ASCII_DAT` format.

Reply:
I think it is fine... easier to understand  it is in-Memory?

Reply:
Yes